articles/model-anatomy.adoc

```diff
@@ -0,0 +1,678 @@
+Typical candidate would be the target's node `name` (if the target node has such a property).
+To support this typical default behavior, LionWeb has a built-in interface `INamed` with one property `name: String` that might be implemented by any classifier.
+However, it's up to the application to select and set the appropriate resolveInfo.
+
+This serves two purposes:
+
+* In the <<interstellar-project, interstellar project example above>>, assume some app that displays the list of external parties for the project as `ESA: 23, JAXA: 42`.
+Now someone thinks there's no project left with ESA, and deletes node 87 (the ESA one).
+If the `externalParties` reference only knew the target node id, the best possible display would be `!unknown target 87!, JAXA: 42` -- not very helpful to the user.
+With resolveInfo, we can display `!unknown target ESA!, JAXA: 42`.
+Now the user has a much better chance of tracking down the error.
+
+* We need to know the target node id in order to store it in a reference.
+But in parser-based systems, after parsing we know only some kind of identifier of a reference target.
+We might want to store this unresolved _abstract syntax tree_ (AST).
+A later linking step hopefully finds the actual target and fills in the proper target node id.
+
+Target node id, resolveInfo, or both must be set for each reference -- otherwise the reference is pointless.
+
+==== Unresolved references
+Assume a node `T` that happens to be the target of a reference `myRef` from node `A`.
+When we delete `T`, the reference stays untouched: `A` still contains a reference `myRef` with target node id `T` (and maybe some resolveInfo), but the target cannot be resolved.
+LionWeb allows references with unresolvable target.
+It's up to the consumer of the model to do something about that target, though.
+
+.Unresolved references example
+[plantuml, deadRef, svg]
+----
+hide empty members
+hide circles
+
+skinparam package {
+  borderColor Transparent
+  backgroundColor Transparent
+}
+
+skinparam class<<h>> {
+  borderColor Transparent
+  backgroundColor Transparent
+  stereotypeFontColor Transparent
+}
+
+package before {
+  class A
+  class T
+  A .> T: myRef
+}
+
+package "after deleting T" {
+  class A
+  class T<<h>>
+  A .> T: myRef
+}
+----
+
+=== Differences between containment and reference
+
+A contained node shares the lifetime of its parent -- if the parent is deleted, the contained node gets deleted, too.
+The contained node makes no sense without its parent.
+Even if we kept the contained node, where would it go?
+We'd need to invent some special place for it.
+In contrast, the target of a reference is perfectly valid without the source.
+They can be deleted independently, and don't need to be deleted when sources pointing to them are deleted.
+
+Containments establish a tree shape for the model.
+With references, we regard the model as a graph, as a node can be target of several references.
+
+.Node graph example
```

Comment:
This graph does not have any explanation and it is not refferred from anywhere in the text.